Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -3184,6 +3220,27 @@ namespace AssetProcessor
                 // Do not process this file yet.  When the job is done it will retrigger processing for this file.
                 return;
             }
+
+            AZ::Outcome<QString, void> topmostSourceAsset = GetFirstTopmostSourceAssetForProductPath(productName);
+
+            if (topmostSourceAsset.IsSuccess())
+            {
+                if (!QFile::exists(topmostSourceAsset.GetValue()))
+                {
+                    AZ_Printf("AP", "Topmost source '%s' for intermediate asset '%.*s' no longer exists, "
+                        "so this intermediate asset will not be reprocessed.",
+                        topmostSourceAsset.GetValue().toUtf8().data(),
+                        AZ_STRING_ARG(productName));
+                    return;
+                }
+            }
+            else
+            {
+                // Failed to find topmost source - don't process this asset, there's a gap or issue in the asset database.
+                // This will hopefully be resolved once all of AP finishes processing.
```

Comment:
nit: "This hopefully..." seems oddly vague and wishful. Is there future work here or a gap that should be closed with another feature or issue.

Reply:
I've altered the language to describe the situation more in detail. Now the "hopefully" can just be read between the lines - if there's a gap there then this will get missed.